inlong-sort/sort-connectors/doris/src/main/java/org/apache/inlong/sort/doris/table/DorisDynamicSchemaOutputFormat.java

```diff
@@ -512,17 +512,12 @@ private void handleDirtyData(Object dirtyData, DirtyType dirtyType, Exception e)
             }
         }
 
-        if (multipleSink && !DirtyType.DESERIALIZE_ERROR.equals(dirtyType) && dirtyData instanceof RowData) {
-            JsonNode rootNode;
-            try {
-                rootNode = jsonDynamicSchemaFormat.deserialize(((RowData) dirtyData).getBinary(0));
-                metricData.outputDirtyMetricsWithEstimate(
-                        jsonDynamicSchemaFormat.parse(rootNode, databasePattern),
-                        null, jsonDynamicSchemaFormat.parse(rootNode, tablePattern), rowSize, dataSize);
-            } catch (Exception ex) {
-                metricData.invokeDirty(rowSize, dataSize);
-            }
-        } else {
+        try {
+            JsonNode rootNode = jsonDynamicSchemaFormat.deserialize(((RowData) dirtyData).getBinary(0));
+            metricData.outputDirtyMetricsWithEstimate(
+                    jsonDynamicSchemaFormat.parse(rootNode, databasePattern),
+                    null, jsonDynamicSchemaFormat.parse(rootNode, tablePattern), rowSize, dataSize);
```

Comment:
rowSize and dataSize should be rowSize = 1 and dataSize is byte length of dirtyData